### PR TITLE
Fixes #19730: API documentation for secret variabel feature

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/19730

### CODE REVIEW

File: webapp/sources/api-doc/code_samples/curl/secret/add.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+add.json:
+
+{
+	"name" : "secret-password",
+	"description" : "Password of my super secret user account",
+  "value" : "123AdminAdmin",
```
Comment (amousset) [inline webapp/sources/api-doc/code_samples/curl/secret/add.sh:6]: incorrect indentation

Comment (amousset) [reply webapp/sources/api-doc/code_samples/curl/secret/add.sh:6]: and maybe use a password that looks like a strong password :smile:

**Comment by ElaadF** [comment]
PR updated with a new commit

File: webapp/sources/api-doc/paths/secret/all.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2021 Normation SAS
+get:
+  summary: List all secrets
+  description: Get the list of all secrets without their value
+  operationId: getAllSecret
```
Comment (ncharles) [inline webapp/sources/api-doc/paths/secret/all.yml:6]: shoudln't it be `getAllSecrets` ?

**Comment by ElaadF** [comment]
PR updated with a new commit

File: webapp/sources/api-doc/paths/secret/all.yml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  tags:
+    - 🧩 Secret Management
+  x-code-samples:
+    - lang: curl
+      source:
+        $ref: ../../code_samples/curl/secret/all.sh
+
+post:
+  summary: Update a secret
+  description: Update a secret and override the value, the name cannot be overrided
```
Comment (amousset) [inline webapp/sources/api-doc/paths/secret/all.yml:61]:
```suggestion
  description: Update a secret and override the value, the name cannot be overridden
```

**Comment by ElaadF** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Pull request merged**